### ISSUE TITLE
Presentation.getManager().getContent() hangs backend after db edit operations

### ISSUE THREAD

**Issue by arnoldas-v** [iTwin/imodel-native#750]
**Describe the bug**
Sometimes after create/update operations `Presentation.getManager().getContent()` call starts responding very slowly or hangs completely. As a result application becomes unusable (no properties can be loaded or no other further edit operations can be started).

**To Reproduce**

Reproducing this issue is very difficult. For normal use it appears seemingly randomly and not on every imodel.
In our application we have a sample tool that would try to force this issue by looping these operations for a few times on specific imodel:
-createElements, saveChanges
-Presentation.getManager().getContent()
-updateElements, saveChanges
-Presentation.getManager().getContent()
-deleteElements, saveChagnes

Every call to getContent() might take multiple minutes or never complete at all.
saveChanges call is very important because without it the issue never happens.

Will provide dumps/steps to build the app and reproduce as needed.

**Expected behavior**
PresentationManager always responds in a reasonable amount of time.

**Screenshots**
![image](https://github.com/iTwin/itwinjs-core/assets/81693692/dfdff0ec-f2d7-4aef-9305-ee083b2ab4fa)
Properties widget timed out and crashed.

**Desktop (please complete the applicable information):**

- OS: windows11
- Electron
- Version 25.8.1
- iTwin.js Version 4.3.3

**Additional context**
Interesting observation:
I noticed that during this hang the whole electron backend becomes not responsive.
For example if you have:
```ts
const timeout () => new Promise(resolve => setTimeout(() => resolve("timed-out"), 1000 * 60 * 1));
await Promise.race([Presentation.getManager().getContent(), timeout()]);
```
The Promise.race will never resolve. Indicating that native does not even give back control to node/electron somehow?

**Comment by grigasp** [comment]
Deadlock call stack:
![image](https://github.com/iTwin/imodel-native/assets/35135765/e8b351a2-791e-4451-babb-1b53fa2995b8)

**Comment by grigasp** [comment]
The fix was released with `itwinjs-core` packages `@4.6.1`

**Comment by GintV** [comment]
@grigasp We're still able to reproduce this issue with Core 4.6.1

**Comment by grigasp** [comment]
> @grigasp We're still able to reproduce this issue with Core 4.6.1

Are you seeing a hang or a slowdown with this version?

**Comment by GintV** [comment]
We got a hang for 10 min for the whole app. After that, properties grid stopped loading properties and we started to get BackendTimeout continuously

**Comment by grigasp** [comment]
> We got a hang for 10 min for the whole app. After that, properties grid stopped loading properties and we started to get BackendTimeout continuously

Any chance you could take a memory dump while its hung?

**Comment by GintV** [comment]
Something like this would help?

https://learn.microsoft.com/en-us/windows-hardware/drivers/debugger/task-manager-live-dump

**Comment by grigasp** [comment]
> Something like this would help? But on our app process, not system as mentioned in the link
> 
> https://learn.microsoft.com/en-us/windows-hardware/drivers/debugger/task-manager-live-dump

Not sure if the ones created through task manager will work. I need a [user-mode dump](https://learn.microsoft.com/en-us/windows-hardware/drivers/debugger/user-mode-dump-files), generally created using Visual studio or WinDbg.

**Comment by grigasp** [comment]
Another fix was released with `4.6.2`, closing the issue for now - please re-open if you can still reproduce the problem after consuming the new version.

**Comment by GintV** [comment]
Thanks for the fixes! Hopefully this will be the last 😃
